torch/csrc/nested_tensor/python_nested_tensor.cpp

```diff
@@ -0,0 +1,362 @@
+    {"grad", (getter)_ListNestedTensorVariable_grad, nullptr, nullptr, nullptr},
+    {"requires_grad",
+     (getter)_ListNestedTensorVariable_requires_grad,
+     nullptr,
+     nullptr,
+     nullptr},
+    {nullptr}};
+
+static PyMethodDef _ListNestedTensorVariable_methods[] = {
+    {"element_size",
+     (PyCFunction)_ListNestedTensorVariable_element_size,
+     METH_NOARGS,
+     "Return element size."},
+    {"nested_dim",
+     (PyCFunction)_ListNestedTensorVariable_nested_dim,
+     METH_NOARGS,
+     "Return nested dim."},
+    {"nested_size",
+     (PyCFunction)_ListNestedTensorVariable_nested_size,
+     METH_NOARGS,
+     "Return nested_size."},
+    {"nested_stride",
+     (PyCFunction)_ListNestedTensorVariable_nested_stride,
+     METH_NOARGS,
+     "Return nested_stride."},
+    {"pin_memory",
+     (PyCFunction)_ListNestedTensorVariable_pin_memory,
+     METH_NOARGS,
+     "Pins memory."},
+    {"detach",
+     (PyCFunction)_ListNestedTensorVariable_detach,
+     METH_NOARGS,
+     "Detaches and returns."},
+    {"requires_grad_",
+     (PyCFunction)_ListNestedTensorVariable_requires_grad_,
+     METH_O,
+     "requires_grad_ and returns."},
+    {"backward",
+     (PyCFunction)_ListNestedTensorVariable_backward,
+     METH_VARARGS,
+     "backward and returns."},
+    {"is_pinned",
+     (PyCFunction)_ListNestedTensorVariable_is_pinned,
+     METH_NOARGS,
+     "Returns is_pinned."},
+    {"is_contiguous",
+     (PyCFunction)_ListNestedTensorVariable_is_contiguous,
+     METH_NOARGS,
+     "Returns is_contiguous."},
+    {"numel",
+     (PyCFunction)_ListNestedTensorVariable_numel,
+     METH_NOARGS,
+     "Returns numel."},
+    {"to_tensor",
+     (PyCFunction)_ListNestedTensorVariable_to_tensor,
+     METH_NOARGS,
+     "Returns to_tensor."},
+    {"dim",
+     (PyCFunction)_ListNestedTensorVariable_dim,
+     METH_NOARGS,
+     "Returns dim."},
+    {"to",
+     (PyCFunction)_ListNestedTensorVariable_to,
+     METH_VARARGS | METH_KEYWORDS,
+     "Returns to."},
+    {"unbind",
+     (PyCFunction)_ListNestedTensorVariable_unbind,
+     METH_NOARGS,
+     "Returns unbound components."},
+    {NULL} /* Sentinel */
+};
+
+static PySequenceMethods _ListNestedTensorVariable_as_sequence = {
+    (lenfunc)_ListNestedTensorVariable_len, /* sq_length */
+    nullptr, /* sq_concat */
+    nullptr, /* sq_repeat */
+    nullptr, /* sq_item */
+    nullptr, /* sq_slice */
+    nullptr, /* sq_ass_item */
+    nullptr, /* sq_ass_slice */
+    nullptr /* sq_contains */
+};
+
+// TODO: "Py_TPFLAGS_DEFAULT enables all memebers defined until Python 3.3"
+// https://docs.python.org/3/extending/newtypes_tutorial.html
+// Does that mean it won't work before Python 3.3?
+PyTypeObject _ListNestedTensorVariableType = {
+    PyVarObject_HEAD_INIT(nullptr, 0) "torch._ListNestedTensor", /* tp_name */
+    sizeof(_ListNestedTensorVariable), /* tp_basicsize */
+    0, /* tp_itemsize */
+    (destructor)_ListNestedTensorVariable_dealloc, /* tp_dealloc */
```

Comment:
won't this -- not work in Python 2?  That seems okay-ish except we haven't actually killed python 2 and the other examples of this don't use this version.

If you want to address those problems first we can keep this, otherwise I'd just copy the existing examples.

Reply:
I'm not sure - should I remove this? What's the right approach?